File: frontend/src/types/park.ts
export type Difficulty = 'easy' | 'medium' | 'hard' | 'expert';

export type PetStatus = 'allowed' | 'restricted' | 'prohibited' | 'unknown';

export type DirectionType = '도보' | '버스' | '셔틀버스' | '지하철' | '자동차' | '기타';

export interface Park {
  id: number;
  name: string;
  lat: number;
  lon: number;
  district: string;
  area: number;
  difficulty: Difficulty;
  avgSlope: number;
  elevationDiff: number;
  petStatus: PetStatus;
}

export interface ParkDetail {
  id: number;
  name: string;
  description: string;
  location: {
    lat: number;
    lon: number;
    district: string;
    address: string;
  };
  information: {
    area: number;
    openedAt: string;
    facilities: Facility[];
    plants: Plant[];
  };
  difficulty: {
    level: Difficulty;
    avgSlope: number;
    elevationDiff: number;
  };
  pet: {
    status: PetStatus;
    notices: string[];
    restrictedLocations: string[];
    serviceAnimalAllowed: boolean;
  };
  notices: string[];
  directions: Direction[];
  contact: {
    department: string;
    phone: string;
    url: string;
  };

  images: {
    image: string;
    map: string;
  };
}

export interface ParkFilter {
  difficulty?: string;
  district?: string;
  petStatus?: string;
}

export interface Facility {
  category: string | null;
  content: string;
}

export interface Plant {
  category: string | null;
  content: string;
}

export interface Direction {
  type: string | null;
  content: string;
}
export interface ParkMap extends Park {
  petRestrictedLocations: string[];
  serviceAnimalAllowed: boolean;
}

export interface ParksResponse<T> {
  items: T[];
  total: number;
}

export interface ParkListResponse extends ParksResponse<Park> {
  page: number;
  pageSize: number;
  totalPages: number;
}

export type ParkMapResponse = ParksResponse<ParkMap>;

File: backend/app/schemas/park.py
from pydantic import BaseModel


class ParkLocation(BaseModel):
    lat: float
    lon: float
    district: str
    address: str


class ParkFacility(BaseModel):
    category: str | None
    content: str


class ParkPlant(BaseModel):
    category: str | None
    content: str


class ParkInformation(BaseModel):
    area: float
    openedAt: str
    facilities: list[ParkFacility]
    plants: list[ParkPlant]


class ParkDifficulty(BaseModel):
    level: str
    avgSlope: float
    elevationDiff: float


class ParkPet(BaseModel):
    status: str
    notices: list[str]
    restrictedLocations: list[str]
    serviceAnimalAllowed: bool


class ParkDirection(BaseModel):
    type: str
    content: str


class ParkContact(BaseModel):
    department: str
    phone: str
    url: str


class ParkImages(BaseModel):
    image: str
    map: str


class ParkListItem(BaseModel):
    id: int
    name: str
    lat: float
    lon: float
    difficulty: str
    avgSlope: float
    elevationDiff: float
    area: float
    district: str
    petStatus: str


class ParkDetailResponse(BaseModel):
    id: int
    name: str
    description: str
    location: ParkLocation
    information: ParkInformation
    difficulty: ParkDifficulty
    pet: ParkPet
    notices: list[str]
    directions: list[ParkDirection]
    contact: ParkContact
    images: ParkImages


class ParkMapItem(BaseModel):
    id: int
    name: str
    lat: float
    lon: float
    difficulty: str
    avgSlope: float
    elevationDiff: float
    area: float
    district: str
    petStatus: str
    petRestrictedLocations: list[str]
    serviceAnimalAllowed: bool


class ParkMapResponse(BaseModel):
    items: list[ParkMapItem]
    total: int


class ParkListResponse(BaseModel):
    items: list[ParkListItem]
    page: int
    pageSize: int
    total: int
    totalPages: int


class ParkSearchResponse(BaseModel):
    items: list[ParkMapResponse]
    total: int
